lib/CL/devices/pthread/pthread.c

```diff
@@ -117,6 +113,10 @@ struct data {
   mem_regions_management* mem_regions;
 #endif
 
+  /* Lock for command list related operations */
+  pthread_mutex_t cq_lock __attribute__ ((aligned (CACHELINE_SIZE)));
```

Comment:
Yes, I meant PTHREAD_OPTIMAL_MUTEX_ALIGNMENT ;) But are you sure it's worth the effort? Did you measure speedups?